### PR TITLE
Update method_formatters.py

### PR DESCRIPTION
**Pull request by tmckenzie51**
removes duplicate block_number_formatter method in favor of to_hex_if_integer method.

### What was wrong?
The methods block_number_formatter and to_hex_if_integer were duplicate functionality.

Related to Issue #1745

### How was it fixed?
block_number_formatter  method was replaced with the to_hex_if_integer method.

### Todo:
- [x] Add entry to the [release notes](https://github.com/ethereum/web3.py/blob/master/newsfragments/README.md)

#### Cute Animal Picture

Red-Tibetan-Mastiff

### CODE REVIEW

**Comment by njgheorghita** [comment]
So, it looks to me like Infura recently updated the version of their IPFS endpoint to `7.0` (running `curl -X POST "https://ipfs.infura.io:5001/api/v0/version"` from your terminal shows the currently supported version). And `py-ipfs-http-client` only supports `0.4.22 ≤ … < 0.7.0`. There are a couple open issues that appear to reference the exact same problem, I've left a comment on one of them. It seems like web3 is blocked until we resolve this, which is definitely not ideal.

Lmk what you think the best way to proceed is, as I see it we have a couple of options.
- Wait until a new version of `py-ipfs-http-client` is released (it's hard to say how soon we can expect this to happen)
- Temporarily skip all tests that use `InfuraIPFSBackend` until `py-ipfs-http-client` is updated
- Reconfigure the tests to use the `LocalIPFSBackend`
- Find a different public gateway we can use. (I've spent a little time trying to connect to a different [public gateway](https://ipfs.github.io/public-gateway-checker/), but have had trouble finding something that works with the [multiaddr](https://github.com/multiformats/multiaddr#string-format) format required by `py-ipfs-http-client`) - imo this is probably the best temporary solution to get us until `py-ipfs-http-client` is updated

**Comment by tmckenzie51** [comment]
> here are a couple open issues that appear to reference the exact same problem, I've left a comment on one of them.

**Comment by tmckenzie51** [comment]
> So, it looks to me like Infura recently updated the version of their IPFS endpoint to `7.0` (running `curl -X POST "https://ipfs.infura.io:5001/api/v0/version"` from your terminal shows the currently supported version). And `py-ipfs-http-client` only supports `0.4.22 ≤ … < 0.7.0`. There are a couple open issues that appear to reference the exact same problem, I've left a comment on one of them. It seems like web3 is blocked until we resolve this, which is definitely not ideal.
> 
> Lmk what you think the best way to proceed is, as I see it we have a couple of options.
> 
> * Wait until a new version of `py-ipfs-http-client` is released (it's hard to say how soon we can expect this to happen)
> * Temporarily skip all tests that use `InfuraIPFSBackend` until `py-ipfs-http-client` is updated
> * Reconfigure the tests to use the `LocalIPFSBackend` (might require a significant chunk of refactoring)
> * Find a different public gateway we can use. (I've spent a little time trying to connect to a different [public gateway](https://ipfs.github.io/public-gateway-checker/), but have had trouble finding something that works with the [multiaddr](https://github.com/multiformats/multiaddr#string-format) format required by `py-ipfs-http-client`) - imo this is probably the best temporary solution to get us until `py-ipfs-http-client` is updated

Okay, I see. Thank you for explaining that. Those sound like decent options that we can consider. @pipermerriam @kclowes  Of these options, which do you think would be the best for us to proceed with?

**Comment by marcgarreau** [comment]
@njgheorghita pointed out a [fix PR](https://github.com/ipfs-shipyard/py-ipfs-http-client/pull/241) waiting for review, but it's not a very active repo. I'm not opposed to just xfail these tests for now, but is that doable for doctests?

**Review by kclowes — APPROVED** [review]
This looks good to me as long as the tests pass after you merge in master!

**Comment by kclowes** [comment]
@tmckenzie51 You can try and push an empty commit to see if that readthedocs failure will work itself out: `git commit --allow-empty -m "Trigger Build"` Let me know if you have questions or want to pair!

**Pull request merged**
